### ISSUE TITLE
"function body not available"?

### ISSUE THREAD

**Issue by soywiz** [v8mips/v8mips#27]
I'm trying to build this project for PSP using the opensource pspsdk.
I'm trying to build it using cygwin.

Executed to install cygwin:

```
wget http://cygwin.com/setup.exe -O cygwin_set.exe
cygwin_set.exe -q -s http://mirrors.kernel.org/sourceware/cygwin --packages gcc --packages make --packages python --packages svn --packages git
```

Downloaded and installed pspsdk:

```
http://sourceforge.net/projects/minpspw/files/SDK%20%2B%20devpak/pspsdk%200.11.2/pspsdk-setup-0.11.2r3.exe/download
```

```
export CC=/cygdrive/c/pspsdk/bin/psp-gcc
export CCX=/cygdrive/c/pspsdk/bin/psp-g++

$ make --version
GNU Make 3.82.90

$ $CC --version
psp-gcc.exe (GCC) 4.3.5
```

Then executed:

```
make dependencies
make mipsel.release
```

The output/error:

```
$ make mipsel.release
GYP_GENERATORS=make \
build/gyp/gyp --generator-output="out" build/all.gyp \
              -Ibuild/standalone.gypi --depth=. \
              -Dv8_target_arch=mipsel \
              -S.mipsel  -Dv8_can_use_vfp3_instructions=true
make[1]: Entering directory `/cygdrive/c/projects/v8m-mips/out'
  CXX(target) /cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/allocation.o
  CXX(target) /cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/atomicops_internals_x86_gcc.o
  CXX(target) /cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/bignum.o
  CXX(target) /cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/bignum-dtoa.o
../src/utils.h: In function `uint64_t v8::internal::double_to_uint64(double)':
../src/utils.h:855: sorry, unimplemented: inlining failed in call to 'Dest v8::internal::BitCast(const Source&) [with Dest = uint64_t, Source = double]': function body not available
../src/bignum-dtoa.cc:37: sorry, unimplemented: called from here
tools/gyp/preparser_lib.target.mipsel.mk:94: recipe for target `/cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/bignum-dtoa.o' failed
make[1]: *** [/cygdrive/c/projects/v8m-mips/out/mipsel.release/obj.target/preparser_lib/src/bignum-dtoa.o] Error 1
make[1]: Leaving directory `/cygdrive/c/projects/v8m-mips/out'
Makefile:183: recipe for target `mipsel.release' failed
make: *** [mipsel.release] Error 2
```

**Comment by kisg** [comment]
Hi,

Which branch did you try to compile? The latest code is on the bleeding_edge branch.

I am pretty sure we never tried to build on Cygwin or using the psp-gcc. Could you try to build on Linux just to eliminate any possible issues that are caused by the Cygwin environment?

I don't know much about the PSP, but from what I have read, it has a MIPS core derived from R4000. I am not sure if it properly implements the MIPS32 ISA. We currently support only MIPS32R1, MIPS32R2 and Loongson. In any case you might have better luck with forcing MIPS32R1 or Loongson ISA using the following GYP settings:

export GYPFLAGS=-Dmips_arch_variant=mips32r1
-or-
export GYPFLAGS=-Dmips_arch_variant=loongson

Best Regards,
Gergely

**Comment by soywiz** [comment]
I'm compiling bleeding_edge.
PSP MIPS is a Allegrex that is R4000 based with some vfpu instructions. (I have created a PSP emulator https://github.com/soywiz/cspspemu and I am trying to compile it and test it there).
I could try to set up the pspsdk enviroment on linux, but since it would take a lot of time, I tried with the precompiled one working on windows.
I have tried both GYPFLAGS without any luck (same error).
A related "bug": http://gcc.gnu.org/bugzilla/show_bug.cgi?id=14677

Regards.
